plugins/modules/storagebox_subaccount.py

```diff
@@ -0,0 +1,473 @@
+extends_documentation_fragment:
+  - community.hrobot.robot
+  - community.hrobot.attributes
+  - community.hrobot.attributes.actiongroup_robot
+
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+  idempotent:
+    support: full
+
+options:
+  storagebox_id:
+    description:
+      - The ID of the storage box to query.
+    type: int
+    required: true
+  password_mode:
+    description:
+      - Controls how password updates are handled.
+      - If C(update-if-provided), the password is updated only if provided (default).
+      - If C(ignore-if-exists), password is only used during creation.
+      - If C(set-to-random), password is reset to a randomly generated one.
+    type: str
+    choices: [update-if-provided, ignore-if-exists, set-to-random]
+    default: update-if-provided
+    required: false
+  subaccount:
+    description:
+      - The subaccount configuration.
+    type: dict
+    required: true
+    suboptions:
+      state:
+        description:
+          - Desired state of this subaccount.
+        choices: [present, absent]
+        default: present
+        type: str
+
+      username:
+        description:
+          - Username of the subaccount.
+          - Required for updating or deleting a subaccount (C(state=absent) or update with C(state=present)).
+          - Omit to create a new subaccount (C(state=present)).
+        type: str
+        required: false
+
+      password:
+        description:
+          - Password to use or change.
+          - If omitted and state is C(present), a random password is generated by hrobot.
+        type: str
+        required: false
+
+      homedirectory:
+        description:
+          - Home directory of the subaccount.
+          - Required when creating or updating a subaccount (C(state=present)).
+        type: str
+        required: false
+
+      samba:
+        description:
+          - Enable or disable Samba.
+        type: bool
+        default: false
+        required: false
+
+      ssh:
+        description:
+          - Enable or disable SSH access.
+        type: bool
+        default: false
+        required: false
+
+      external_reachability:
+        description:
+          - Enable or disable external reachability.
+        type: bool
+        default: false
+        required: false
+
+      webdav:
+        description:
+          - Enable or disable WebDAV.
+        type: bool
+        default: false
+        required: false
+
+      readonly:
+        description:
+          - Enable or disable read-only mode.
+        type: bool
+        default: false
+        required: false
+
+      comment:
+        description:
+          - A custom comment for the subaccount.
+        type: str
+        default: ""
+        required: false
+      idempotence:
+        description:
+          - Select which attribute to use to check subaccount existence.
+          - C(username) or C(comment).
```

Comment:
The choices are already listed because of `choices`, so only list them again if there are explanations.
```suggestion
          - If set to C(username), then subaccounts are identified by their username.
            Note that usernames cannot be specified on creation, so you need to use different
            module arguments for creation and updating.
          - If set to C(comment). then subaccounts are identified by their comment.
            If there already exist more than one subaccount with the given comment, the module will fail.
```